go/cmd/dolt/commands/diff.go

```
@@ -560,36 +536,146 @@ func diffRows(ctx context.Context, newRows, oldRows types.Map, newSch, oldSch sc
+	allCols := sch.GetAllCols()
+	err := untypedUnionSch.GetAllCols().Iter(func(tag uint64, col schema.Column) (stop bool, err error) {
+		col, ok := allCols.GetByTag(tag)
+
+		if ok {
+			tagToCol[tag] = col.Name
+		} else {
+			tagToCol[tag] = ""
+		}
+
+		return false, nil
+	})
+
+	if err != nil {
+		return nil, errhand.BuildDError("error: failed to map columns to tags").Build()
+	}
+
+	return tagToCol, nil
+}
+
+func createSplitter(newSch schema.Schema, oldSch schema.Schema, joiner *rowconv.Joiner) (schema.Schema, *diff.DiffSplitter, errhand.VerboseError) {
+	dumbNewSch, err := dumbDownSchema(newSch)
+
+	if err != nil {
+		return nil, nil, errhand.BuildDError("").AddCause(err).Build()
+	}
+
+	dumbOldSch, err := dumbDownSchema(oldSch)
+
+	if err != nil {
+		return nil, nil, errhand.BuildDError("").AddCause(err).Build()
+	}
+
+	untypedUnionSch, err := untyped.UntypedSchemaUnion(dumbNewSch, dumbOldSch)
+
+	if err != nil {
+		return nil, nil, errhand.BuildDError("Failed to merge schemas").Build()
+	}
+
+	newToUnionConv := rowconv.IdentityConverter
+	if newSch != nil {
+		newToUnionMapping, err := rowconv.TagMapping(newSch, untypedUnionSch)
+
+		if err != nil {
+			return nil, nil, errhand.BuildDError("Error creating unioned mapping").AddCause(err).Build()
+		}
+
+		newToUnionConv, _ = rowconv.NewRowConverter(newToUnionMapping)
+	}
+
+	oldToUnionConv := rowconv.IdentityConverter
+	if oldSch != nil {
+		oldToUnionMapping, err := rowconv.TagMapping(oldSch, untypedUnionSch)
+
+		if err != nil {
+			return nil, nil, errhand.BuildDError("Error creating unioned mapping").AddCause(err).Build()
+		}
+
+		oldToUnionConv, _ = rowconv.NewRowConverter(oldToUnionMapping)
+	}
+
+	ds := diff.NewDiffSplitter(joiner, oldToUnionConv, newToUnionConv)
+	return untypedUnionSch, ds, nil
 }
 
 var emptyHash = hash.Hash{}
 
 func printTableDiffSummary(tblName string, tbl1, tbl2 *doltdb.Table) {
 	bold := color.New(color.Bold)
 
-	bold.Printf("diff --dolt a/%[1]s b/%[1]s\n", tblName)
```

Trying to shut up warnings about ignored errors here.